python/pyarrow/array.pxi

```diff
@@ -1990,6 +1990,12 @@ cdef class DictionaryArray(Array):
     def dictionary_encode(self):
         return self
 
+    def dictionary_decode(self):
+        """
+        Decodes the DictionaryArray to an Array.
+        """
+        return self.dictionary.take(self.indices)
+
```

Comment:
Just a thought, since the base `Array` has a `dictionary_encode` method and the `DictionaryArray` overrides it with a noop, should we also add a corresponding noop `dictionary_decode` to the base `Array`?

Reply:
I thought about that, but I realised it didn't make much sense to expose a `dictionary_decode` method in `Array` which is not dictionary encoded.
On the opposite a `dictionary_encode` method makes sense on `Array` as it's something that can be dictionary encoded.